### PR TITLE
Gradle plugin: Create bundle task from a Jar task

### PR DESCRIPTION
**Pull request by vierbergenlars**
Add an utility method to the `alfrescoDynamicExtensions` extension object to apply the DeBundleConvention to arbitrary Jar tasks and create an InstallBundle task for them as well.

This update drops support for Gradle versions < 5.3 (March 2019, so two years old), which is why I marked it as breaking change. I'm not sure that we should regard it as a breaking change for all of Dynamic Extensions, since you can just keep using an older version of the plugin.

## Description

This will make it easier to create a Dynamic Extensions jar from an arbitrary Jar task.

```groovy
task someJar(type: Jar) { } // Pretend that this is created by some Gradle plugin that you can't modify

alfrescoDynamicExtensions.configureBundle(tasks.someJar)

someJar {
    bnd([
        // ...
    ])
}
```

Next to the `someJar` task that will now build as a Dynamic Extensions jar, an `installSomeBundle` task is also automatically created. It will install the `someJar` task in the configured Alfresco repository.

## Motivation and Context

In some projects, I have to create a [shadowJar](https://imperceptiblethoughts.com/shadow) to shade dependencies into a Dynamic Extensions jar.
Unfortunately, this means that I can't make use of the convenient configuration that the DE Gradle plugin performs for me, and I have to do some crazy operations to make my shadowed jar compatible with Dynamic Extensions.

<details>
<summary>Example</summary>

```groovy
task shadowJar(type: ShadowJar) {
    classifier = "shadowjar"
    from(sourceSets.main.output)
    configurations = [project.configurations.shadowed]
}

jar {
    from(project.provider({ zipTree(shadowJar.outputs.files.singleFile) }))
    dependsOn shadowJar

    bnd(
            'Bundle-Name': project.description,
            'Bundle-Description': project.description,
            'Bundle-SymbolicName': "${project.group}.${project.name}",
            'Alfresco-Spring-Configuration': 'eu.xenit.testing.integrationtesting',
            'Import-Package': 'org.alfresco.*,org.springframework.*,com.github.dynamicextensionsalfresco.*,org.osgi.*,org.slf4j.*',
            'Export-Package': 'junit.framework;version=4.12,'
                    + 'org.junit;version=4.12,'
                    + 'org.junit.runner;version=4.12,'
                    + 'org.junit.runner.manipulation;version=4.12,'
                    + 'org.junit.runners;version=4.12,'
                    + 'org.junit.runners.model;version=4.12,'
                    + 'org.junit.experimental.categories;version=4.12,'
                    + 'org.junit.internal;version=4.12,'
                    + 'org.junit.rules;version=4.12,'
                    + 'org.junit.matchers;version=4.12,'
                    + 'org.junit.runner.notification;version=4.12,'
                    + "eu.xenit.testing.integrationtesting.runner.*;version=${project.version.version.normal}"
    )
}
```
</details>

## How Has This Been Tested?

Integration test in `gradle-plugin` subproject.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change) 

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the changelog.

### CODE REVIEW

**Comment by kerkhofsd** [comment]
build says no

**Review by kerkhofsd — APPROVED** [review]
Nice!

File: gradle-plugin/build.gradle [hunk 1]
```diff
@@ -5,6 +5,7 @@ plugins {
     id "java-gradle-plugin"
     id "groovy"
     id "maven-publish"
+    id "jacoco"
```
Comment (kerkhofsd) [inline gradle-plugin/build.gradle:8]: Does that mean we shouldn't ignore the failing code coverage :upside_down_face: ?

**Review by kerkhofsd — APPROVED** [review]

File: gradle-plugin/build.gradle (hunk 1 above)
Comment (vierbergenlars) [reply gradle-plugin/build.gradle:8]: At least that was my attempt.
But sonarqube does not seem to pick up any coverage for the plugin code :disappointed:

**Pull request merged**